### PR TITLE
Add CircleCI workflows

### PR DESCRIPTION
**Pull request by paulosman**
Just run the tests nightly and on each commit

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Golang CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-go/ for more details
+version: 2.1
+
+orbs:
+  buildevents: honeycombio/buildevents@0.2.6
```
Comment (martin308) [inline .circleci/config.yml:7]: if this is a public repo then we can't use buildevents

**Review by martin308 — CHANGES_REQUESTED** [review]
It's probably worth using the `golang` orb to make this simpler https://circleci.com/developer/orbs/orb/circleci/go

File: .circleci/config.yml (hunk 1 above)
Comment (MikeGoldsmith) [reply .circleci/config.yml:7]: This is because buildevents needs a private secret to pass the honeycomb API Key, so will fail on forks 👍

Comment (paulosman) [reply .circleci/config.yml]: :+1: Removed

**Comment by paulosman** [comment]
> It's probably worth using the `golang` orb to make this simpler https://circleci.com/developer/orbs/orb/circleci/go

Yeah, that is much better. Also added go 1.15!

**Review by MikeGoldsmith — APPROVED** [review]

**Review by martin308 — APPROVED** [review]
Looks good, could probably use a matrix for the go versions but with just two it's not a biggie

**Pull request merged**